src/components/CheckIn/CheckInItems.js

```diff
@@ -0,0 +1,218 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import ReactRouterPropTypes from 'react-router-prop-types';
+import {
+  get,
+} from 'lodash';
+
+import {
+  Button,
+  Callout,
+  Checkbox,
+  Col,
+  MultiColumnList,
+  Row,
+  SearchField,
+} from '@folio/stripes/components';
+
+import {
+  RECEIVING_HISTORY,
+  // ORDER_PIECES,
+} from '../Utils/resources';
+import { LIMIT_MAX } from '../Utils/const';
+// import AddPieceModal from './AddPieceModal';
+
+class CheckInItems extends Component {
+  static manifest = Object.freeze({
+    query: {},
+    RECEIVING_HISTORY,
+    // ORDER_PIECES,
+  })
+
+  static propTypes = {
+    mutator: PropTypes.object.isRequired,
+    match: ReactRouterPropTypes.match.isRequired,
+    // stripes: PropTypes.object.isRequired,
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    // this.connectedAddPieceModal = props.stripes.connect(AddPieceModal);
+    this.callout = React.createRef();
+  }
+
+  state = {
+    // addPieceModalOpened: false,
+    isAllChecked: false,
+    items: [],
+    searchText: '',
+  };
+
+  componentDidMount() {
+    this.fetchItems();
+  }
+
+  fetchItems = () => {
+    const { mutator, match: { params: { id, lineId } } } = this.props;
+    const params = {
+      limit: LIMIT_MAX,
+      query: `checkin == true and purchaseOrderId==${id}${lineId ? ` and poLineId==${lineId}` : ''}`,
+    };
+
+    mutator.RECEIVING_HISTORY.reset();
+    mutator.RECEIVING_HISTORY.GET({ params })
+      .then(items => {
+        this.setState({ items });
+      });
+  }
+
+  getItems = () => {
+    const { items, searchText } = this.state;
+
+    return searchText
+      ? items.filter(item => item.title.toLocaleLowerCase().includes(searchText))
+      : items;
+  }
+
+  toggleItem = (item) => {
+    item.isChecked = !item.isChecked;
+    this.setState(state => {
+      const items = [...state.items];
+
+      return { items };
+    });
+  }
+
+  toggleAll = () => {
+    this.setState((state) => {
+      const isAllChecked = !state.isAllChecked;
+      const items = state.items.map(item => ({
+        ...item,
+        isChecked: isAllChecked,
+      }));
+
+      return {
+        isAllChecked,
+        items,
+      };
+    });
+  }
+
+  changeSearchText = (event) => {
+    const searchText = get(event, 'target.value', '').toLocaleLowerCase();
+
+    this.setState({ searchText });
+  }
+
+  // addPieceModalOpen = () => {
```

Comment:
are these comments for future use?

Reply:
yes, they are going to be uncommented in the next PR, so they will still be green to review